### ISSUE TITLE
The wallets does not show up in the bottom sheet and show grey screen instead

### ISSUE THREAD

**Issue by shahji-mufin** [WalletConnect/Web3ModalFlutter#65]
**Describe the bug**
```
Fatal Exception: io.flutter.plugins.firebase.crashlytics.FlutterError: RangeError (end): Invalid value: Not in inclusive range 0..2: 4. Error thrown building Widget.
       at ListBase.getRange(dart:collection)
       at _WalletsListShortPageState.build.<fn>(wallets_list_short_page.dart:58)
       at _ExplorerServiceItemsListenerState.build.<fn>.<fn>(explorer_service_items_listener.dart:43)
       at _ValueListenableBuilderState.build(value_listenable_builder.dart:186)
       at StatefulElement.build(framework.dart:5409)
       at ComponentElement.performRebuild(framework.dart:5297)
       at StatefulElement.performRebuild(framework.dart:5462)
       at Element.rebuild(framework.dart:5016)
       at BuildOwner.buildScope(framework.dart:2779)
       at WidgetsBinding.drawFrame(binding.dart:916)
       at RendererBinding._handlePersistentFrameCallback(binding.dart:360)
       at SchedulerBinding._invokeFrameCallback(binding.dart:1297)
       at SchedulerBinding.handleDrawFrame(binding.dart:1227)
       at SchedulerBinding._handleDrawFrame(binding.dart:1085)
```

**To Reproduce**
1. Initialize WalletConnect SDK as Getit service'
2. Launch the app

**Expected behavior**
No error to be seen of such kind

**Screenshots**
![image (1)](https://github.com/WalletConnect/Web3ModalFlutter/assets/113975718/fc449802-b224-4472-83d5-1e17c880ba5b)

**Smartphone (please complete the following information):**
 - Device: [OnePluse 8 Pro, Oppo F9, Xiaomi Mi A1]
 - OS: [ Android 13, Android 10, Android 11]

**Additional context**

**Comment by shahji-mufin** [comment]
This issue is the same as [includedWalletIds option causing an exception](https://github.com/WalletConnect/Web3ModalFlutter/issues/56)

**Comment by quetool** [comment]
Hello @shahji-mufin
That issue has been solved here https://github.com/WalletConnect/Web3ModalFlutter/pull/58
Are you using the latest version of the SDK?
Can I have a min reproducible code?

**Comment by shahji-mufin** [comment]
Hi @quetool, thanks for the quick response. 
Yes, I saw the issue [#58](https://github.com/WalletConnect/Web3ModalFlutter/pull/58) but it is still coming up.
I am using `web3modal_flutter: ^3.0.0-beta16` and now shifted to `web3modal_flutter: ^3.0.0`
I have made some changes and tested it further, if the issue still persists, I will comment here otherwise I will close the ticket.
The funny thing is no error came on Android 14 😅
